### PR TITLE
SCC-2109 is-research optimizations

### PR DESCRIPTION
**Pull request by mwbenowitz**
This creates several optimizations in the `is-research` service. These both cover performance concerns as well as maintainability. The high level things this PR does include:

- Removing the AWS Lambda Layer configuration. This should improve maintainability by simplifying the repo and making ongoing development and testing much easier
- Moving the calls to get the mixed bibs list and NYPL Core values to the initialization stage, which should provide a small performance boost as they are not loaded on each invocation.
- Reordering of the `is-research?` checks as if an item is from a partner or is a mixed bib its status can be determined without making an API call
- Reorganize events and add coverage tool to the test suite
- Add configuration for a development environment

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -10,6 +10,12 @@ def init
   $logger = NyplLogFormatter.new(STDOUT, level: ENV['LOG_LEVEL'] || 'info')
   $platform_api = PlatformApiClient.new
 
+  $nypl_core = NyplCore.new
```
Comment (siberry) [inline app.rb]: This is also on line 9

File: spec/unit/app_spec.rb [hunk 2]
```diff
@@ -28,25 +30,37 @@
...
   end
 
-  xit "should respond with a 404 if requested item isn't found" do
-    event = JSON.parse(File.read("./event-not_found.json"))
+  it "should respond with a 404 if requested item isn't found" do
+    event = JSON.parse(File.read("./events/event-not_found.json"))
     response = handle_event(event: event, context: '')
 
-    lamba_resp = JSON.parse(response[:body])
+    lambda_resp = JSON.parse(response[:body])
```
Comment (siberry) [inline spec/unit/app_spec.rb:60]: Ha! thanks

**Review by siberry — APPROVED** [review]
This is great! Only one small comment. Otherwise, these are all great changes, especially moving the call to the `items` endpoint.

File: app.rb (hunk 1 above)
Comment (mwbenowitz) [reply app.rb]: good catch, thanks!

**Pull request merged**